tap_klaviyo_custom/streams.py

```diff
@@ -1,157 +1,122 @@
-    #Defining the url_base outside of the class results in an error
-    url_base = 'https://a.klaviyo.com/api/v2/'
+    schema_filepath = SCHEMAS_DIR / "lists.json"
 
-    records_jsonpath = "$.records[*]"
+    @property
+    def partitions(self) -> Optional[List[dict]]:
+        """Get stream partitions."""
+        return [{"list_id": id} for id in self.config["list_ids"]]
 
-    def get_url_params(self, partition: Optional[dict]) -> Dict[str, Any]:
-        """Return a dictionary of values to be used in URL parameterization."""
-        params = {}
-        params.update({"api_key": self.config['api_key']})
-        return params
-
-    def prepare_request(
-        self, context: Optional[dict], next_page_token: Optional[Any], list_id: Optional[str]
-    ) -> requests.PreparedRequest:
-        """Prepare a request object.
-
-        If partitioning is supported, the `context` object will contain the partition
-        definitions. Pagination information can be parsed from `next_page_token` if
-        `next_page_token` is not None.
-        """
-        http_method = self.rest_method
-        url: str = self.get_url(context, list_id)
-        params: dict = self.get_url_params(context)
-        request_data = self.prepare_request_payload(context, next_page_token)
-        headers = self.http_headers
-
-        if next_page_token != None:
-            params['marker'] = next_page_token
-
-        request = cast(
-            requests.PreparedRequest,
-            self.requests_session.prepare_request(
-                requests.Request(
-                    method=http_method,
-                    url=url,
-                    params=params,
-                    headers=headers,
-                    json=request_data,
-                )
-            ),
-        )
-        # sleep timer to avoid Klaviyo API rate limit errors
-        time.sleep(1)
-        return request
+    def get_next_page_token(
+        self, response: requests.Response, previous_token: Optional[Any]
+    ) -> Any:
+        """Return token identifying next page or None if all records have been read."""
+        # since we're using partitions, there's only ever one page per partition
+        return None
 
-    def get_url(self, context: Optional[dict], list_id: Optional[str]) -> str:
-        """Return a URL, optionally targeted to a specific partition or context.
+    def post_process(self, row: dict, context: Optional[dict] = None) -> Optional[dict]:
+        """As needed, append or transform raw data to match expected structure."""
+        # For some reason,  the id is not included in the response
+        assert context is not None
+        return {**row, "list_id": context["list_id"]}
 
-        Developers override this method to perform dynamic URL generation.
-        """
-        url = self.url_base+self.path.format(list_id=list_id)
+    def get_child_context(self, record: dict, context: Optional[dict]) -> dict:
```

Comment:
If ListsStream is not the Parent Stream of ListMembersStream, what's the point of get_child_context() and doing the partition as part of ListsStream rather than ListMembersStream?

Reply:
ListsStream *is* the parent of ListMembersStream. Together `get_child_context` and `parent_stream_type` do this